### PR TITLE
Tile Map Service Catalog Item

### PR DESCRIPTION
**Pull request by keattang**
Pretty self explanatory

### CODE REVIEW

**Comment by kring** [comment]
Thanks!  By the way, have you seen Cesium's new `UrlTemplateImageryProvider`?  It might be better to create a catalog item for that rather than TMS (and we can get rid of the `OpenStreetMapCatalogItem`, too).  The only thing we'd lose is the TMS provider's inspection of `tilemapresource.xml`.  Is that useful for you guys?

**Comment by keattang** [comment]
We don't use the metadata anywhere. Does the imagery provider use it to optimise the tiles it calls by not calling tiles that are outside the bounds? Will UrlTemplateImageryProvider be able to do the same?

**Comment by kring** [comment]
Yes it uses the metadata (if available) to get min/max levels, tiling scheme, valid extent, etc.  All of these things can be specified to `UrlTemplateImageryProvider` explicitly, it just won't read them automatically from `tilemapresource.xml`.  It sounds like `UrlTemplateImageryProvider` might be what you want.

File: lib/Models/TileMapServiceCatalogItem.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+     * @type {String}
+     */
+    this.url = undefined;
+
+    /**
+     * Gets or sets the maximum zoom level.
+     * @type {Integer}
+     * @default 18
+     */
+    this.maxZoom = 18;
```
Comment (kring) [inline lib/Models/TileMapServiceCatalogItem.js:37]: Is this line the cause of #939?

**Comment by kring** [comment]
@keattang sorry for sitting on this for so long.  Is it still relevant?  As mentioned above, I think `UrlTemplateCatalogItem` might be the better way to go.  I'm going to close it but feel free to reopen if you think it still makes sense.

**Comment by keattang** [comment]
@kring I'm looking at implementing a `UrlTemplateCatalogItem` for TMS data but I still want to use the metadata included in `tilemapresource.xml` so as to avoid requests to tiles that don't exist. What would you recommend is the best way to do this? Should I inspect `tilemapresource.xml` separately and feed the info into the `UrlTemplateCatalogItem` or should I allow some provision in the catalog item to try go and find the metadata?

**Pull request closed without merging**